server/services/offlinePayment.service.js

```
@@ -118,3 +119,112 @@ export const getAllOfflinePayments = async (user, { page = 1, limit = 10, ...que
   const result = offlinePayments[0].data;
   return { pagination, result };
 };
+
+export const resolveOfflinePayment = async ({ offlinePaymentId, adminId }) => {
+  const offlinePayment = await getOfflinePaymentById(offlinePaymentId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+  if (!offlinePayment) {
+    throw new ErrorHandler(httpStatus.NOT_FOUND, 'Invalid offline payment');
+  }
+
```

I think we need to add one more thing here 🙏🏽 . We need to check if the admin has resolved all the comments before resolving it? Since we are dealing with money here, let's throw an error here if the offline payment still has unresolved comments when resolving.

Of course, you need to add tests for this too 😺 . Thank you.